### PR TITLE
feat: Add `mlflow-ai-gw-image` module and update mlflow to 2.17.0

### PR DESCRIPTION
**Pull request by dlpzx**
## Describe your changes
Add `mlflow-ai-gw-image` module that builds mlflow Docker image and pushes to ECR.
The image hosts a [MLFlow AI Gateway server](https://mlflow.org/docs/latest/llms/index.html#id2).

Next steps:
We need to decide how we want to use the Docker image. We could put it in a Lambda+API Gateway to really have an AI API Gateway. Or we could link it to the rest of the modules of mlflow and deploy it as an ECS task

## Checklist before requesting a review

- [x] I updated `CHANGELOG.MD` with a description of my changes
- [x] If the change was to a module, I ran the code validation script (`scripts/validate.sh`)
- [ ] If the change was to a module, I have added thorough tests ---> just CodeBuild, no actual cdk app
- [x] If the change was to a module, I have added/updated the module's README.md
- [x] If a module was added, I added a reference to the module to the repository's README.md
- [X] I verified that my code deploys successfully using `seedfarmer apply`

-[X] I pulled the image from ECR and then built the server locally

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: c87c905796209c5a1552458b4170b4832ccdd5d7
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ce9f16d2-a94a-49d0-a81b-99f297010b6f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/mlflow/mlflow-ai-gw-image/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Mlflow Image
+
+## Description
+
+This module creates an mlflow tracking server container image and pushes to the specified Elastic Container Repository.
```
Comment (kukushking) [inline modules/mlflow/mlflow-ai-gw-image/README.md:5]: nit: copy-paste

**Review by kukushking — APPROVED** [review]
Looks great, thank you!

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: f6181b2db28123d9ef26701d1c2242fb53a63335
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=dc1007d4-7511-4919-be65-dede7d1b22e4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: eddfd92e9a47a81bf08944b9c88066b476265c37
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=0bcf382e-f584-40bf-a17a-79320bb9e351%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: a2901df2e4a458cec5b1f680e089d1ca1803bb70
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=cbadf4ee-0f5d-4d86-a287-23a05e8836f6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]
Good to go - thanks!

**Pull request merged**